keras_nlp/src/utils/preset_utils.py

```diff
@@ -192,6 +208,41 @@ def get_file(preset, path):
         )
 
 
+def copy_gfile_to_cache(fname, url, cache_subdir):
+    """Much of this is adapted from get_file of keras core."""
+    if url is None:
+        raise ValueError(
+            'Please specify the "url" argument (URL of the file '
+            "to download)."
+        )
+
+    if cache_subdir is None:
+        cache_dir = config.keras_home()
+
+    datadir_base = os.path.expanduser(cache_dir)
+    if not os.access(datadir_base, os.W_OK):
+        datadir_base = os.path.join("/tmp", ".keras")
+    datadir = os.path.join(datadir_base, cache_subdir)
+    os.makedirs(datadir, exist_ok=True)
+
+    fname = path_to_string(fname)
+    if not fname:
+        fname = os.path.basename(urllib.parse.urlsplit(url).path)
+        if not fname:
+            raise ValueError(
+                "Can't parse the file name from the origin provided: "
+                f"'{url}'."
+                "Please specify the `fname` as the input param."
+            )
+    fpath = os.path.join(datadir, fname)
+
+    if not os.path.exists(fpath):
+        io_utils.print_msg(f"Downloading data from {url}")
```

Comment:
Use https://github.com/keras-team/keras-nlp/blob/5a734f9926af0d215246f4bc47738281e1006aba/keras_nlp/src/utils/keras_utils.py#L91, we copied this to avoid using private API from Keras (which can change without warning)